crates/red_knot_python_semantic/resources/mdtest/type_properties/is_subtype_of.md

```diff
@@ -456,5 +456,257 @@ static_assert(not is_subtype_of(Intersection[Unknown, int], int))
+
+If the corresponding parameter in the subtype is a keyword-only parameter, it behaves in the same
+way. This is because keyword-only parameter is one of the kind of standard parameter.
+
+```py
+from knot_extensions import CallableTypeFromFunction, is_subtype_of, static_assert
+
+def standard_a(a: int) -> None: ...
+def keyword_b(*, b: int) -> None: ...
+
+static_assert(not is_subtype_of(CallableTypeFromFunction[standard_a], CallableTypeFromFunction[keyword_b]))
+
+def standard_float(a: float) -> None: ...
+def keyword_int(*, a: int) -> None: ...
+
+static_assert(is_subtype_of(CallableTypeFromFunction[standard_float], CallableTypeFromFunction[keyword_int]))
+
+def standard_with_default(a: int = 1) -> None: ...
+def keyword_with_default(*, a: int = 1) -> None: ...
+def empty() -> None: ...
+
+static_assert(is_subtype_of(CallableTypeFromFunction[standard_with_default], CallableTypeFromFunction[keyword_with_default]))
+static_assert(is_subtype_of(CallableTypeFromFunction[standard_with_default], CallableTypeFromFunction[empty]))
+```
+
+And, the same is for positional-only parameter except that the names are not required to be the
+same.
+
+```py
+from knot_extensions import CallableTypeFromFunction, is_subtype_of, static_assert
+
+def standard_a(a: int) -> None: ...
+def positional_b(b: int, /) -> None: ...
+
+static_assert(is_subtype_of(CallableTypeFromFunction[standard_a], CallableTypeFromFunction[positional_b]))
+
+def standard_float(a: float) -> None: ...
+def positional_int(a: int, /) -> None: ...
+
+static_assert(is_subtype_of(CallableTypeFromFunction[standard_float], CallableTypeFromFunction[positional_int]))
+
+def standard_with_default(a: int = 1) -> None: ...
+def positional_with_default(a: int = 1, /) -> None: ...
+def empty() -> None: ...
+
+static_assert(is_subtype_of(CallableTypeFromFunction[standard_with_default], CallableTypeFromFunction[positional_with_default]))
+static_assert(is_subtype_of(CallableTypeFromFunction[standard_with_default], CallableTypeFromFunction[empty]))
+```
+
+And, with other kinds of parameter:
+
+```py
+def variadic(*args: int) -> None: ...
+def keyword_variadic(**kwargs: int) -> None: ...
+
+static_assert(not is_subtype_of(CallableTypeFromFunction[standard_a], CallableTypeFromFunction[variadic]))
+static_assert(not is_subtype_of(CallableTypeFromFunction[standard_a], CallableTypeFromFunction[keyword_variadic]))
+```
+
+#### Variadic
+
+The name of the variadic parameter does not need to be the same in the subtype.
+
+```py
+from knot_extensions import CallableTypeFromFunction, is_subtype_of, static_assert
+
+def variadic_float(*args2: float) -> None: ...
+def variadic_int(*args1: int) -> None: ...
+
+static_assert(is_subtype_of(CallableTypeFromFunction[variadic_float], CallableTypeFromFunction[variadic_int]))
+static_assert(not is_subtype_of(CallableTypeFromFunction[variadic_int], CallableTypeFromFunction[variadic_float]))
+```
+
+A variadic parameter can be omitted in the subtype:
```

Comment:
Don't you mean "omitted in the supertype"?

Reply:
Yes 🤦